docs/vcpkg-schema-definitions.schema.json

```diff
@@ -0,0 +1,715 @@
+        }
+      },
+      "required": [
+        "packages"
+      ]
+    },
+    "package-name": {
+      "type": "string",
+      "description": "Name of a package.",
+      "allOf": [
+        {
+          "description": "Package name must be a dot-separated list of valid identifiers",
+          "pattern": "^[a-z0-9]+(-[a-z0-9]+)*(\\.[a-z0-9]+(-[a-z0-9]+)*)*$"
+        },
+        {
+          "not": {
+            "description": "Identifiers must not be a Windows filesystem or vcpkg reserved name.",
+            "pattern": "(^|\\.)(prn|aux|nul|con|lpt[1-9]|com[1-9]|core|default)(\\.|$)"
+          }
+        }
+      ]
+    },
+    "platform-expression": {
+      "description": "A specification of a set of platforms. See https://github.com/microsoft/vcpkg/blob/master/docs/specifications/manifests.md#definitions.",
+      "type": "string"
+    },
+    "port-version": {
+      "description": "A non-negative integer indicating the port revision. If this field doesn't exist, it's assumed to be `0`.",
+      "type": "integer",
+      "minimum": 0,
+      "default": 0
+    },
+    "registry-array": {
+      "oneOf": [
+        {
+          "allOf": [
+            {
+              "oneOf": [
+                {
+                  "$ref": "#/definitions/builtin-registry"
+                },
+                {
+                  "$ref": "#/definitions/git-registry"
+                },
+                {
+                  "$ref": "#/definitions/filesystem-registry"
+                }
+              ]
+            },
+            {
+              "$ref": "#/definitions/packages-list"
+            }
+          ]
+        },
+        {
+          "$ref": "#/definitions/artifact-registry"
+        }
+      ]
+    },
+    "semantic-version": {
+      "description": "A semantic version string. See https://semver.org/",
+      "type": "string",
+      "pattern": "^(0|[1-9]\\d*)\\.(0|[1-9]\\d*)\\.(0|[1-9]\\d*)(?:-((?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\\.(?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\\+([0-9a-zA-Z-]+(?:\\.[0-9a-zA-Z-]+)*))?$"
+    },
+    "sha1": {
+      "type": "string",
+      "pattern": "[0-9a-fA-F]{40}"
+    },
+    "sha256": {
+      "type": "string",
+      "pattern": "[0-9a-fA-F]{64}"
+    },
+    "sha512": {
+      "type": "string",
+      "pattern": "[0-9a-fA-F]{128}"
+    },
+    "slash-identifier": {
+      "type": "string",
+      "description": "Slash identifiers are used for artifact names.",
+      "allOf": [
+        {
+          "description": "Slash identifiers are lowercase with digits, dashes, and forward slashes.",
+          "pattern": "[a-z0-9]+([/-][a-z0-9]+)*"
+        },
+        {
+          "not": {
+            "description": "Identifiers and parts delimited by slashes must not be a Windows filesystem or vcpkg reserved name.",
+            "pattern": "(^|/)(prn|aux|nul|con|lpt[1-9]|com[1-9]|core|default)(/|$)"
+          }
+        }
+      ]
+    },
+    "string-or-strings": {
+      "if": {
```

Comment:
Why use this construction over `"oneOf"` (or vice-versa)?

Reply:
If I recall correctly this form resulted in better intellisense behavior since with oneOf the validator tries to validate both paths and while it doesn't know what the intended one is, you get confusing messages. We should probably consider making them all use the if form but I don't want to block over that.